test/UnitTest/Microsoft.Test.AspNet.OData/Formatter/Serialization/ODataResourceSerializerTests.cs

```diff
@@ -1315,7 +1311,11 @@ public void CreateODataAction_IncludesEverything_ForFullMetadata()
             ODataAction actualAction = _serializer.CreateODataAction(action, context);
 
             // Assert
+#if NETCORE
```

Comment:
Can you add a note as to why these should be different?

Reply:
I'm unsure why we'd want to use different routes in AspNetCore vs AspNet. The more the test cases differ, the more likely we are to accidentally regress the product.

Reply:
We can't just new up a UrlHelper like before, it needs to come from the container. That is why HttpContextExtensions.cs works like that.

I understand the change to request factory.

My questions is: Why should the AspNet test use an HttpRoute and AspNetCore use an ODataRoute?

Maybe we can find a way to make the Router in request factory use AspNetCore's Route instead of ODataRoute so the test response is the same?

Or make the AspNet test use an OData route?